tests/test_jobqueue.py

```diff
@@ -283,6 +283,66 @@ def test_run_daily_with_timezone(self, job_queue):
         assert self.result == 1
         assert job_queue._queue.get(False)[0] == pytest.approx(expected_reschedule_time)
 
+    def test_run_monthly(self, job_queue):
+        delta, now = 0.1, time.time()
+        date_time = dtm.datetime.utcfromtimestamp(now)
+        time_of_day = (date_time + dtm.timedelta(seconds=delta)).time()
+        expected_reschedule_time = now + delta
+
+        day = date_time.day
+        next_job_day = date_time + dtm.timedelta(calendar.monthrange(
+                                                 date_time.year, date_time.month)[1])
+        expected_reschedule_time += calendar.monthrange(date_time.year,
+                                                        date_time.month)[1] * 24 * 60 * 60
+        while next_job_day.day != day:
```

Comment:
Why is this still needed? should be covered by the above lines?

Reply:
yups you're right, I can replace this `while` loop with an `if` instead. We still need to do this check to cover the case whereby the next month does not have a particular date. 

For example, if the test runs on Jan 31st, the next job should be run on March 31st instead of Feb, since Feb does not have this date